blockchain/chain_sync/src/sync.rs

```diff
@@ -99,14 +101,36 @@ where
         db: Arc<DB>,
         network_send: Sender<NetworkMessage>,
         network_rx: Receiver<NetworkEvent>,
+        genesis_buffer: Option<BufReader<File>>,
     ) -> Result<Self, Error> {
         let chain_store = ChainStore::new(db.clone());
         let genesis = match chain_store.genesis()? {
             Some(gen) => Tipset::new(vec![gen])?,
             None => {
                 // TODO change default logic for genesis or setup better initialization
-                warn!("No genesis found in data storage, using a default");
-                Tipset::new(vec![BlockHeader::default()])?
+                warn!("no genesis found in data storage, trying to load from file");
+                match genesis_buffer {
+                    Some(buf) => {
```

Comment:
Also related to this, but shouldn't the genesis file option through the CLI supersede the genesis in storage? Seems weird that if a genesis exists in storage, that this will just be quietly ignored

Reply:
Hmm, I suppose youre right about that. I can make that happen